src/components/DataTable.vue

```diff
@@ -0,0 +1,208 @@
+  data: {type: Array, required: true},
+  allowExport: {type: Boolean, default: true}
+});
+
+let selectedRows = [];
+const refSelectedRows = ref(selectedRows);
+
+const exportCSV = async () => {
+  const csv = Papa.unparse(_flatMap(refSelectedRows.value, prepareData));
+  const blob = new Blob([csv]);
+  const a = document.createElement('a');
+  a.href = URL.createObjectURL(blob, { type: 'text/plain' });
+  a.download = 'runs.csv';
+  document.body.appendChild(a);
+  a.click();
+  document.body.removeChild(a);
+};
+
+const exportFullCSV = async () => {
+  const csv = Papa.unparse(_flatMap(refData.value, prepareData))
+  const blob = new Blob([csv]);
+  const a = document.createElement('a');
+  a.href = URL.createObjectURL(blob, { type: 'text/plain' });
+  a.download = 'runs.csv';
+  document.body.appendChild(a);
+  a.click();
+  document.body.removeChild(a);
+};
+
+// Generate filters and options objects
+const valid_dataTypes = ['NUMERIC', 'TEXT', 'DATE'];
+let filters = {};
+let options = {};
+_forEach(props.columns, column => {
+  // Check if header text is supplied; if not, generate.
+  if(!_get(column, 'header')){
+    column['header'] = _startCase(_get(column, 'field'))
+  }
+  const dataType = _toUpper(_get(column, 'dataType'));
+  let returnMatchMode = null;
+  if(valid_dataTypes.includes(dataType)){
+    if(dataType === 'NUMERIC'){
+      returnMatchMode = { value: null, matchMode: FilterMatchMode.EQUALS};
+    } else if(dataType === 'TEXT'){
+      returnMatchMode = { value: null, matchMode: FilterMatchMode.STARTS_WITH};
+    } else if(dataType === 'DATE'){
+      returnMatchMode = { value: null, matchMode: FilterMatchMode.DATE_IS};
+    }
+    
+    if(_get(column, 'useMultiSelect')){
+      returnMatchMode = { value: null, matchMode: FilterMatchMode.IN};
+      options[column.field] = getUniqueOptions(column);
+    }
+  }
+  if(_get(column, 'allowMultipleFilters') === true && returnMatchMode){
+    filters[column.field] = {
+      operator: FilterOperator.AND,
+      constraints: [returnMatchMode]
+    }
+  } else if(returnMatchMode) {
+    filters[column.field] = returnMatchMode;
+  }
+})
+const refOptions = ref(options);
+const refFilters = ref(filters);
+
+// Grab list of fields defined as dates
+let dateFields = _filter(props.columns, col => _toUpper(col.dataType) === 'DATE');
+dateFields = _map(dateFields, col => col.field);
+
+let computedData = _forEach(props.data, entry => {
+  // Clean up date fields to use Date objects
+  _forEach(dateFields, field => {
+    let dateEntry = _get(entry, field);
+    if(dateEntry !== null) entry[field] = new Date(dateEntry);
+  })
+})
+const refData = ref(computedData);
+
+// Generate list of options given a column
+function getUniqueOptions(column){
+  const field = _get(column, 'field');
+  let options = [];
+  _forEach(props.data, entry => {
+    if(!options.includes(entry[field])){
+      options.push(entry[field]);
+    }
+  });
+  return options
+}
+
+function prepareData(entry){
```

Comment:
```suggestion
function prepareDates(entry){
```

Perhaps this function name would be more explicit about what the function is doing.